src/staticModelAdapt.cpp

```diff
@@ -0,0 +1,128 @@
+                    f_b = f_m;
+                } else{
+                    a = m;
+                    f_a = f_m;
+                }
+                m = (a+b)/2.0;
+                f_m = CESSdiff(logweight,loglike,m-curr,desiredCESS);
+                err = abs(f_m);
+            }
+            return m;
+        }
+    }
+
+    /// Chooses the next temperature such that a desired conditional ESS is maintained.
+    ///
+    /// \param logweight An armadillo vector containing the logarithm of the current particle weights.
+    /// \param loglike An armadillo vector containing the log likelihood of the current particle values.
+    /// \param desiredCESS The target conditional ESS for the next temperature (generally fixed).
+    void staticModelAdapt::ChooseTemp(const arma::vec & logweight, const arma::vec & loglike, double desiredCESS) {
+        double temp_curr = temp.back();
+        if (CESSdiff(logweight,loglike,1-temp_curr,desiredCESS)>=0){
+            temp.push_back(1.0);
+        } else {
+            temp.push_back(bisection(temp_curr, logweight, loglike, desiredCESS));
+        }
+    }
+
+    /// Calculates the empirical covariance matrix based on the current weighted particle set.
+    ///
+    /// \param theta An [Nxd] armadillo matrix of doubles for the current particle values, where N is
+    /// the number of particles and d is the dimension of the parameter.
+    /// \param logweight An armadillo vector containing the logarithm of the current particle weights.
+    void staticModelAdapt::calcEmpCov(const arma::mat & theta, const arma::vec & logweight){
+        int N = logweight.n_rows;
+        arma::vec normWeights = exp(logweight - stableLogSumWeights(logweight));
+
+        arma::mat diff = theta - arma::ones(N,1)*arma::mean(theta,0);
+        empCov = diff.t()*diagmat(normWeights)*diff;
+    }
+
+    /// Calculates the Cholesky decomposition of the empirical covariance matrix based on the current weighted particle set.
+    ///
+    /// \param theta An [Nxd] armadillo matrix of doubles for the current particle values, where N is
+    /// the number of particles and d is the dimension of the parameter
+    /// \param logweight An armadillo vector of the logarithm of the current particle weights
+    void staticModelAdapt::calcCholCov(const arma::mat & theta, const arma::vec logweight){
+        calcEmpCov(theta,logweight);
+        cholCov = arma::chol(empCov);
+    }
+
+    /// Calculates the number of MCMC repeats based on the results of the most recent set of MCMC moves.
+    ///
+    /// \param acceptProb The proportion of accepted MCMC steps in the most recent iteration.
+    /// \param desiredAcceptProb The desired probability of a successful move for each particle.
+    /// \param initialN The initial number of MCMC repeats.
+    /// \param maxReps The maximum number of MCMC repeats.
+    int staticModelAdapt::calcMcmcRepeats(double acceptProb, double desiredAcceptProb, int initialN, int maxReps){
```

Comment:
It's a bad idea to compare doubles with integers using ==, usually (the finite precision issue), might it make sense to replace these comparisons with 
abs(acceptProb - 1) <= 10**-9
or similar?